python/triton/language/semantic.py

```diff
@@ -1418,8 +1418,9 @@ def where(condition: tl.tensor, x: tl.tensor, y: tl.tensor, builder: ir.builder)
         condition, x = broadcast_impl_value(condition, x, builder)
         x, y = broadcast_impl_value(x, y, builder)
         condition, x = broadcast_impl_value(condition, x, builder)
-
-    x, y = binary_op_type_checking_impl(x, y, builder, True, True)
+    # Bypass arithmetic type check for FP8 types where they are not supported.
+    is_fp8 = x.type == y.type and x.type.is_fp8() and y.type.is_fp8()
+    x, y = binary_op_type_checking_impl(x, y, builder, True, True, not is_fp8)
```

Comment:
so fp8 would not have auto-promote but other types would? Seems a bit odd. I would add support for fp8 in `binary_op_type_checking_impl` instead

Reply:
The current logics does not support fp8 because fp8 arithmetic is not available on hardware (except for dot). Do I understand it correctly?

Also, I'm not sure we need promote here, as in the example kernel fp8 is not used for arithmetic, rather, the kernel loads fp32, convert it to fp8, and conditionally stored out.

Reply:
Or do you think it's safe to not promote anything when x.type == y.type?

Reply:
I meant in general if we have a mix-mode op we promote to the highest format. We could do that for fp8 as well right?

Reply:
Oh yeah we should do that.